chainer/functions/evaluation/classification_summary.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import division
+
+import numpy
+import six
+
+import chainer
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _f1_score(precision, recall, beta):
+    beta_square = beta * beta
+    return ((1 + beta_square) * precision * recall /
+            (beta_square * precision + recall)).astype(precision.dtype)
+
+
+class ClassificationSummary(function.Function):
+
+    ignore_label = -1
+
+    def __init__(self, beta, label_num):
```

Comment:
I found an additional problem; it seems that ``ignore_label`` option is missing.